### PR TITLE
Terminal preferences UI

### PR DESCRIPTION
**Pull request by uranusjr**
This supersedes and closes #737. See that pull request for detailed description.

### CODE REVIEW

File: MacDown/Code/Preferences/MPTerminalPreferencesViewController.m [hunk 1]
```diff
@@ -152,8 +152,8 @@ - (void)installShellUtility
     NSFileManager *fm = [NSFileManager defaultManager];
     if ([fm fileExistsAtPath:utilityBundlePath])
     {
-        BOOL ok = [fm copyItemAtPath:utilityBundlePath
-                              toPath:MPCommandInstallationPath error:NULL];
+        BOOL ok = [fm createSymbolicLinkAtPath:MPCommandInstallationPath
+                           withDestinationPath:utilityBundlePath error:NULL];
```
Comment (niklasberglund) [inline MacDown/Code/Preferences/MPTerminalPreferencesViewController.m]: Perfect. I assumed symlinking wasn't possible through `NSFileManager` but happy to be wrong

File: MacDown/Code/Utility/MPHomebrewSubprocessController.m [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+- (instancetype)init
+{
+    return [self initWithArguments:nil];
+}
+
+- (void)runWithCompletionHandler:(void(^)(NSString *))handler
+{
+    _completionHandler = handler;
+    @try {
+        [_task launch];
```
Comment (niklasberglund) [inline MacDown/Code/Utility/MPHomebrewSubprocessController.m]: I'm nitpicking here but what I've seen elsewhere in the codebase is that `self.` syntax has been used for accessing properties, even properties declared in .m file. E.g. in MPMathJaxListener.m, MPPlugin.m, MPGeneralPreferencesViewController.m etc.

File: MacDown/Code/Utility/MPHomebrewSubprocessController.m [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+- (instancetype)init
+{
+    return [self initWithArguments:nil];
+}
+
+- (void)runWithCompletionHandler:(void(^)(NSString *))handler
+{
+    _completionHandler = handler;
+    @try {
```
Comment (niklasberglund) [inline MacDown/Code/Utility/MPHomebrewSubprocessController.m]: Should this { be on a separate line? Does { on new line for `if`, `for` etc from the MacDown coding style(_CONTRIBUTING.md_) also apply to @try and @catch?

File: MacDown/Code/Preferences/MPTerminalPreferencesViewController.m [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        [sharedSupportURL URLByAppendingPathComponent:@"bin/macdown"].path;
+
+    NSFileManager *fm = [NSFileManager defaultManager];
+    if ([fm fileExistsAtPath:utilityBundlePath])
+    {
+        BOOL ok = [fm createSymbolicLinkAtPath:MPCommandInstallationPath
+                           withDestinationPath:utilityBundlePath error:NULL];
+        if (ok)
+            [self lookForShellUtility];
+        // TODO: Handle removal failure.
```
Comment (niklasberglund) [inline MacDown/Code/Preferences/MPTerminalPreferencesViewController.m]: "Handle removal failure" => "Handle install failure"

File: MacDown/Code/Utility/MPHomebrewSubprocessController.m [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        if (handler)
+            handler(nil);
+    }
+}
+
+- (void)homebrewReadDidComplete:(NSNotification *)notification
+{
+    NSFileHandle *handle = notification.object;
+
+    // We don't need this anymore.
```
Comment (niklasberglund) [inline MacDown/Code/Utility/MPHomebrewSubprocessController.m]: At first I thought this comment meant this is a code block we don't need so it should be removed. Then I read the code and got that it's removing an observer we don't need any more.

Comment (uranusjr) [inline MacDown/Code/Utility/MPHomebrewSubprocessController.m]: I should rewrite this to make it clear.

File: MacDown/Code/Preferences/MPTerminalPreferencesViewController.m (hunk 4 above)
Comment (uranusjr) [inline MacDown/Code/Preferences/MPTerminalPreferencesViewController.m]: Copy-paste FAIL

File: MacDown/Code/Utility/MPHomebrewSubprocessController.m (hunk 2 above)
Comment (uranusjr) [inline MacDown/Code/Utility/MPHomebrewSubprocessController.m]: You’re right. I was swinging between the two styles when implementing class. I decided to use `self.` but missed this.

File: MacDown/Code/Utility/MPHomebrewSubprocessController.m (hunk 3 above)
Comment (uranusjr) [inline MacDown/Code/Utility/MPHomebrewSubprocessController.m]: Right.

BTW, I wanted to rewrite this with more idiomatic error handling (with `NSError *`), but couldn’t find a way. Is there any?

Comment (niklasberglund) [inline MacDown/Code/Utility/MPHomebrewSubprocessController.m]: @uranusjr I don't know any way to achieve that other than wrapping it

**Pull request merged**